pynapple/core/metadata_class.py

```diff
@@ -0,0 +1,242 @@
+import numpy as np
+import pandas as pd
+
+
+class MetadataBase:
```

Comment:
I changed to the second suggestion for now, unless you think changing to an abstract class is better

Reply:
I am ok with this. Maybe @BalzaniEdoardo has a different opinion about this one?

Reply:
I would define an abstract class only if we think we need to subclass it and the subclass must reimplement some methods.

For example, if we have a Tsd vs TsGroup metadata method that must differ in the implementation. Otherwise private is fine.